.github/workflows/publish_snapshot.yml

```diff
@@ -0,0 +1,95 @@
+          properties: 'opensearch_hadoop'
+
+      - name: publish snapshots to build repositories
+        run: |
+          export SONATYPE_USERNAME=$(aws secretsmanager get-secret-value --secret-id maven-snapshots-username --query SecretString --output text)
+          export SONATYPE_PASSWORD=$(aws secretsmanager get-secret-value --secret-id maven-snapshots-password --query SecretString --output text)
+          echo "::add-mask::$SONATYPE_USERNAME"
+          echo "::add-mask::$SONATYPE_PASSWORD"
+          ./gradlew :opensearch-hadoop-mr:publishMainPublicationToBuildRepository
+          ./gradlew :opensearch-hadoop-hive:publishMainPublicationToBuildRepository
+          ./gradlew -S -Dbuild.snapshot=true -Dorg.gradle.warning.mode=summary distribution
+
+      - name: create snapshots maven folder and copy files
+        run: |
+          mkdir -p maven/org/opensearch/client
+          mkdir -p maven/org/opensearch/client/opensearch-hadoop/${{ steps.read_property.outputs.opensearch_hadoop }}
+          mkdir -p maven/org/opensearch/client/opensearch-spark-20_2.10/${{ steps.read_property.outputs.opensearch_hadoop }}
+          mkdir -p maven/org/opensearch/client/opensearch-spark-20_2.11/${{ steps.read_property.outputs.opensearch_hadoop }}
+          mkdir -p maven/org/opensearch/client/opensearch-spark-20_2.12/${{ steps.read_property.outputs.opensearch_hadoop }}
+          mkdir -p maven/org/opensearch/client/opensearch-spark-30_2.12/${{ steps.read_property.outputs.opensearch_hadoop }}
+          mkdir -p maven/org/opensearch/client/opensearch-spark-30_2.13/${{ steps.read_property.outputs.opensearch_hadoop }}
+          cp -R mr/build/repo/org/opensearch/client/ maven/org/opensearch/client/
+          cp -R hive/build/repo/org/opensearch/client/ maven/org/opensearch/client/
+          cp -R dist/build/distributions/ maven/org/opensearch/client/opensearch-hadoop/${{ steps.read_property.outputs.opensearch_hadoop }}
+          cp -R spark/sql-20/build/distributions/opensearch-spark-20_2.10* maven/org/opensearch/client/opensearch-spark-20_2.10/${{ steps.read_property.outputs.opensearch_hadoop }}
+          cp -R spark/sql-20/build/distributions/opensearch-spark-20_2.11* maven/org/opensearch/client/opensearch-spark-20_2.11/${{ steps.read_property.outputs.opensearch_hadoop }}
+          cp -R spark/sql-20/build/distributions/opensearch-spark-20_2.12* maven/org/opensearch/client/opensearch-spark-20_2.12/${{ steps.read_property.outputs.opensearch_hadoop }}
+          cp -R spark/sql-30/build/distributions/opensearch-spark-30_2.12* maven/org/opensearch/client/opensearch-spark-30_2.12/${{ steps.read_property.outputs.opensearch_hadoop }}
+          cp -R spark/sql-30/build/distributions/opensearch-spark-30_2.13* maven/org/opensearch/client/opensearch-spark-30_2.13/${{ steps.read_property.outputs.opensearch_hadoop }}
+
+      - uses: actions/checkout@v3
+        with:
+          repository: 'opensearch-project/opensearch-build'
```

Comment:
Please use https://github.com/opensearch-project/opensearch-build-libraries/blob/main/resources/publish/publish-snapshot.sh as this will be the one up to date. The one in build repo was recently added back due to 1.x build issues.

Reply:
Using the new one from the build-libraries repo.